signer/core/apitypes/types.go

```diff
@@ -490,13 +500,34 @@ func parseInteger(encType string, encValue interface{}) (*big.Int, error) {
 func (typedData *TypedData) EncodePrimitiveValue(encType string, encValue interface{}, depth int) ([]byte, error) {
 	switch encType {
 	case "address":
-		stringValue, ok := encValue.(string)
-		if !ok || !common.IsHexAddress(stringValue) {
-			return nil, dataMismatchError(encType, encValue)
+		stringValue, isString := encValue.(string)
+		if isString {
+			if !common.IsHexAddress(stringValue) {
+				return nil, dataMismatchError(encType, encValue)
+			}
+			retval := make([]byte, 32)
+			copy(retval[12:], common.HexToAddress(stringValue).Bytes())
+			return retval, nil
+		}
+		byteSlice, isByteSlice := encValue.([]byte)
+		if isByteSlice {
+			if len(byteSlice) == 20 {
+				retval := make([]byte, 32)
+				copy(retval[12:], byteSlice)
+				return retval, nil
+			} else if len(byteSlice) == 32 {
+				return byteSlice, nil
```

Comment:
Why would it be 32 bytes long? Addresses have a size of 20 bytes.

Reply:
Removed support for this, thought it may be useful.